src/Plugins/Tools/ReportGenerator/ReportGenerator.cpp

```diff
@@ -0,0 +1,133 @@
+#include "ReportGenerator.h"
+#include "App/AppDirs.h"
+#include "App/AppLog.h"
+#include "rgTelemetry/TelemetryAccessor.h"
+#include "rgTelemetry/TelemetryResolver.h"
+#include "rgUtils/TemplateStepper.h"
+#include <Telemetry/LookupTelemetry.h>
+#include <Telemetry/Telemetry.h>
+#include <Telemetry/TelemetryShare.h>
+#include <Vehicles/Vehicles.h>
+#include <QDebug>
+#include <QDesktopServices>
+#include <QtCore>
+
+namespace RG {
+ReportGenerator::ReportGenerator(Fact *parent)
+    : Fact(parent,
+           QString(PLUGIN_NAME).toLower(),
+           tr("Report generator"),
+           tr("Report creation using templates"),
+           Group,
+           "file-document")
+    , m_template_loaded(false)
+{
+    f_open_browser = new Fact(this,
+                              "enable",
+                              tr("Open Browser"),
+                              tr("Open generated file in Browser"),
+                              Fact::Bool,
+                              "web");
+
+    f_choose_template = new Fact(this,
+                                 "template-picker",
+                                 tr("Select template"),
+                                 tr(""),
+                                 Action | Apply,
+                                 "upload");
+
+    connect(f_choose_template, &Fact::triggered, this, &ReportGenerator::loadTemplateSlot);
+
+    f_generate_report = new Fact(this,
+                                 "generate_report",
+                                 tr("Generate report"),
+                                 tr(""),
+                                 Action | Apply,
+                                 "export");
+    connect(f_generate_report, &Fact::triggered, this, &ReportGenerator::generateReportSlot);
+
+    f_generate_report->setEnabled(false);
+}
+
+void ReportGenerator::generateReportSlot()
+{
+    if (!m_template_loaded)
+        return;
+
+    generateReport();
+    saveReport();
+}
+
+void ReportGenerator::loadTemplate()
+{
+    m_template_loaded = false;
```

Comment:
Instead of using additional bool you can just check m_template_raw for emptiness.